spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestCreateBranch.java

```diff
@@ -129,6 +129,24 @@ public void testCreateBranchUseCustomMaxSnapshotAge() throws NoSuchTableExceptio
     Assert.assertNull(ref.maxRefAgeMs());
   }
 
+  @Test
+  public void testCreateBranchIfNotExists() throws NoSuchTableException {
+    long maxSnapshotAge = 2L;
+    Table table = createDefaultTableAndInsert2Row();
+    String branchName = "b1";
+    sql(
+        "ALTER TABLE %s CREATE BRANCH %s WITH SNAPSHOT RETENTION %d DAYS",
+        tableName, branchName, maxSnapshotAge);
+    sql("ALTER TABLE %s CREATE BRANCH IF NOT EXISTS %s", tableName, branchName);
+
+    table.refresh();
+    SnapshotRef ref = table.refs().get(branchName);
+    Assert.assertNotNull(ref);
```

Comment:
```suggestion
    Assert.assertEquals(table.currentSnapshot().snapshotId(), ref.snapshotId());
```

Reply:
Good point, we can have better test hardness that way to validate the actual snapshot